### DISCUSSION TITLE
Is it possible to call the <Web3NetworkSwitch /> component predefining the network to be switched to?

### DISCUSSION THREAD

**Discussion by aromeoes** [WalletConnect/walletconnect-monorepo#2344] [Developer Support] [answered]
I'm using v2 with React

I'm having issues understanding if there is documentation for interacting with the modal at more depth than that at https://docs.walletconnect.com/2.0/web3modal/react/components

I was expecting a hook that allowed me to provide this experience. Am I supposed to use WAGMI hooks instead?
The reason I was trying to use <Web3NetworkSwitch /> is that it automatically tries to add the network for the user's wallet if it doesn't exist

**Comment by glitch-txs** [comment, marked as answer]
You can set a [default chain](https://docs.walletconnect.com/2.0/web3modal/options#defaultchain-optional) for this. You can also use all wagmi's hooks you need.
